### PR TITLE
explorer: Remove all network references

### LINKED ISSUE #2630: explorer: Remove all references to `node` and create URLs based on the current URL
The application no longer needs references to the node URL or network to connect to as this information can be taken from the subdomain of the application.  All env var references to host node proxies should be removed and, instead, constructed in the application logic.

### PR DESCRIPTION
**Pull request by kieranhall**
Resolves #2630

### CODE REVIEW

File: explorer/src/lib/components/__tests__/__snapshots__/Navbar.spec.js.snap [hunk 1]
```diff
@@ -44,7 +44,7 @@ exports[`Navbar > renders the Navbar component 1`] = `
     <span
       class="dusk-badge dusk-badge--variant-success dusk-navbar__menu--network"
     >
-      Local
+      nodes.dusk.network
```
Comment (deuch13) [inline explorer/src/lib/components/__tests__/__snapshots__/Navbar.spec.js.snap:47]: Is this what we want to display in the Navbar badge? I am asking because I do not know if we are moving away from `Local, Testnet, Devnet, Mainnet`.

File: explorer/src/routes/+page.svelte [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
-  import { onNetworkChange } from "$lib/lifecyles";
 
   const pollingDataStore = createPollingDataStore(
     duskAPI.getLatestChainInfo,
     $appStore.fetchInterval
   );
 
-  onNetworkChange((network) => {
+  onMount(() => {
     pollingDataStore.reset();
```
Comment (deuch13) [inline explorer/src/routes/+page.svelte:16]: See above about the `reset()`

File: explorer/src/routes/blocks/+page.svelte [hunk 3]
```diff
@@ -3,32 +3,27 @@
...
+  import { onDestroy, onMount } from "svelte";
 
   const pollingDataStore = createPollingDataStore(
     duskAPI.getBlocks,
     $appStore.fetchInterval
   );
 
-  onNetworkChange((network) => {
+  onMount(() => {
     pollingDataStore.reset();
```
Comment (deuch13) [inline explorer/src/routes/blocks/+page.svelte:14]: Same as above

File: explorer/src/routes/blocks/block/+page.svelte [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
   const payloadStore = createDataStore(duskAPI.getBlockDetails);
 
   const getBlock = () => {
-    dataStore.getData($appStore.network, $page.url.searchParams.get("id"));
-    payloadStore.getData($appStore.network, $page.url.searchParams.get("id"));
+    dataStore.getData($page.url.searchParams.get("id"));
+    payloadStore.getData($page.url.searchParams.get("id"));
   };
 
   const updateData = () => {
```
Comment (deuch13) [inline explorer/src/routes/blocks/block/+page.svelte:17]: `updateData` can be removed and the `onMount` hook can directly call `getBlock`

File: explorer/src/routes/transactions/+page.svelte [hunk 5]
```diff
@@ -3,33 +3,27 @@
...
+  import { onDestroy, onMount } from "svelte";
 
   const pollingDataStore = createPollingDataStore(
     duskAPI.getTransactions,
     $appStore.fetchInterval
   );
 
-  onNetworkChange((network) => {
+  onMount(() => {
     pollingDataStore.reset();
```
Comment (deuch13) [inline explorer/src/routes/transactions/+page.svelte:14]: See above about `reset()`

File: explorer/src/routes/transactions/transaction/+page.svelte [hunk 6]
```diff
@@ -20,7 +20,7 @@
     getTransaction();
   };
 
-  onNetworkChange(updateData);
+  onMount(updateData);
```
Comment (deuch13) [inline explorer/src/routes/transactions/transaction/+page.svelte:23]: Similar to the `block` route. The `updateData` fn can be removed and `getTransaction` can be directly called in the `onMount` hook.

File: explorer/src/routes/blocks/block/+page.svelte (hunk 4 above)
Comment (deuch13) [reply explorer/src/routes/blocks/block/+page.svelte:17]: After looking at this closer the `onMount` hook and `getBlock` can be removed and directly call 
`
dataStore.getData($page.url.searchParams.get("id"));
payloadStore.getData($page.url.searchParams.get("id"));
`

File: explorer/src/lib/components/__tests__/__snapshots__/Navbar.spec.js.snap (hunk 1 above)
Comment (kieranhall) [reply explorer/src/lib/components/__tests__/__snapshots__/Navbar.spec.js.snap:47]: I was considering taking the first part of the domain name, perhaps that is better now that the network names are no longer configured.

File: explorer/src/lib/url/makeNodeUrl.js [hunk 7]
```diff
@@ -1,37 +1,40 @@
...
     default: // localnet
       node = new URL(
-        `${import.meta.env.VITE_RUSK_PATH || "/"}`,
+        `${import.meta.env.VITE_RUSK_PATH || "/"}${path}`,
         import.meta.url
       );
 
       if (import.meta.env.VITE_NODE_URL) {
-        node = new URL(import.meta.env.VITE_NODE_URL, import.meta.url);
+        node = new URL(
```
Comment (ascartabelli) [inline explorer/src/lib/url/makeNodeUrl.js:34]: I see this is not part of this PR, but I don't get why you're redeclaring `node` here: shouldn't this be an "if / else" maybe?

File: explorer/src/lib/services/__tests__/duskAPI.spec.js [hunk 8]
```diff
@@ -25,14 +25,13 @@ describe("duskAPI", () => {
   };
 
   /** @type {URL} */
-  const gqlExpectedURL = new URL(`/02/Chain`, node.origin);
-
+  const gqlExpectedURL = new URL(`/02/Chain`, node);
```
[the remark this answers is not in the document]
Comment (kieranhall) [reply explorer/src/lib/services/__tests__/duskAPI.spec.js:28]: Should be an error for this, you're right.

[the remark this answers is not in the document]
Comment (ascartabelli) [reply explorer/src/lib/services/__tests__/duskAPI.spec.js:28]: Maybe the rule was conflicting with something that prettier does and was disabled?

**Review by ascartabelli — APPROVED** [review]

**Review by ascartabelli — APPROVED** [review]
